core/domain/collection_services.py

```diff
@@ -134,23 +134,34 @@ def get_collection_from_model(collection_model, run_conversion=True):
     """
 
     # Ensure the original collection model does not get altered.
-    versioned_collection = {
+    versioned_collection_content = {
         'schema_version': collection_model.schema_version,
-        'nodes': copy.deepcopy(collection_model.nodes)
+        'collection_content': copy.deepcopy(collection_model.collection_content)
     }
 
+    # If collection_content is empty, attempt to retrieve nodes data from nodes
+    # instead. This is temporary, and intended to not break backwards
+    # compatibility before the migration job is run.
+    # TODO(wxy): Remove this after collection migration is completed.
```

Comment:
We can, actually, if it's dead code. They could upgrade piecemeal (to release X, first, do the migration, and then upgrade again to release Y). Similar to answer migration.